SW4STM32/BellaLui/Application/HostBoard/Src/misc/Test/state_machine_tests.cpp

```diff
@@ -0,0 +1,140 @@
+#include "misc/state_machine_helpers.h"
+
+#include "gtest/gtest.h"
+#include "misc/rocket_constants.h"
+
+TEST(StateMachineTests, ShouldStayInLiftoffStateDuringMotorBurnTime){
+    //ASSERT_EQ(handleLiftoffState(ROCKET_CST_MOTOR_BURNTIME - 10, 0), STATE_LIFTOFF);
+    ASSERT_FALSE(state_machine_helpers::handleLiftoffState(ROCKET_CST_MOTOR_BURNTIME - 10, 0));
+}
+
+TEST(StateMachineTests, ShouldSwitchFromLiftoffToCoastStateAfterMotorBurnTime){
+    //ASSERT_EQ(handleLiftoffState(ROCKET_CST_MOTOR_BURNTIME + 10, 0), STATE_COAST);
+    ASSERT_TRUE(state_machine_helpers::handleLiftoffState(ROCKET_CST_MOTOR_BURNTIME + 10, 0));
+}
+
+TEST(StateMachineTests, ShouldGetCorrectImuStatus){
+    uint8_t imuIsReady = state_machine_helpers::newImuDataIsAvailable(4, 3);
+    EXPECT_EQ(imuIsReady, 1);
+    imuIsReady = state_machine_helpers::newImuDataIsAvailable(3,3);
+    EXPECT_EQ(imuIsReady, 0);
+    imuIsReady = state_machine_helpers::newImuDataIsAvailable(2,3);
+    EXPECT_EQ(imuIsReady, 0);
+}
+
+TEST(StateMachineTests, ShouldGetCorrectBarometerStatus){
+    uint8_t barometerIsReady = state_machine_helpers::newBarometerDataIsAvailable(4, 3);
+    EXPECT_EQ(barometerIsReady, 1);
+    barometerIsReady = state_machine_helpers::newBarometerDataIsAvailable(3,3);
+    EXPECT_EQ(barometerIsReady, 0);
+    barometerIsReady = state_machine_helpers::newBarometerDataIsAvailable(2,3);
+    EXPECT_EQ(barometerIsReady, 0);
+}
+
+TEST(StateMachineTests, ShouldReachTouchdownStateAfterFiveMinutes){
+    uint32_t currentTime = 2*60*1000;
+    uint32_t liftoffTime = 1*60*1000;
+    EXPECT_FALSE(state_machine_helpers::touchdownStateIsReached(currentTime, liftoffTime));
+    currentTime = 8*60*1000;
+    EXPECT_TRUE(state_machine_helpers::touchdownStateIsReached(currentTime, liftoffTime));
+    liftoffTime = 0;
+    EXPECT_FALSE(state_machine_helpers::touchdownStateIsReached(currentTime, liftoffTime));
+}
+
+TEST(StateMachineTests, ShouldGetCorrectIdleStateStatus){
+    uint32_t currentTime = 2*60*1000;
+    uint32_t liftoffTime = 1*60*1000;
+    float acceleration_z = 1;
+
+    uint8_t state_idle_status = state_machine_helpers::handleIdleState(currentTime, liftoffTime, acceleration_z);
+    EXPECT_EQ(state_idle_status, state_machine_helpers::state_idle_false_positive);
+
+    acceleration_z = 2.1;
+    state_idle_status = state_machine_helpers::handleIdleState(currentTime, liftoffTime, acceleration_z);
+    EXPECT_EQ(state_idle_status, state_machine_helpers::state_idle_switch_to_liftoff_state);
+
+    currentTime = liftoffTime + 300;
+    state_idle_status = state_machine_helpers::handleIdleState(currentTime, liftoffTime, acceleration_z);
+    EXPECT_EQ(state_idle_status, state_machine_helpers::state_idle_no_op);
+
+    liftoffTime = 0;
+    state_idle_status = state_machine_helpers::handleIdleState(currentTime, liftoffTime, acceleration_z);
+    EXPECT_EQ(state_idle_status, state_machine_helpers::state_idle_liftoff_detected);
+}
+
```

Comment:
Just noticed that a test for the coast state is missing for some reason. will add it asap